apache/trafficcontrol
# Validating absolute DNS name requirement on Static DNS Entry Type

## What does this PR (Pull Request) do?


- [x] This PR fixes a bug in Traffic_Portal's UI and Traffic_Ops API callback. Github Issue: #4743 (https://github.com/apache/trafficcontrol/issues/4743)

## Which Traffic Control components are affected by this PR?


- Traffic Portal
- Traffic Ops

## What is the best way to verify this PR?

**Manual Testing** 
1. Create a HTTP delivery service
2. Click on More and then add Static DNS entry
3. Select a Type:CNAME_RECORD and type address without a trailing period. On clicking create, you should see an error stating: `address must be a valid DNS entry. Missing a trailing period at the end.`
4. Once you do add a trailing period, you can create a static DNS entry with a successful message.

**Automation Testing**
1. Run CIAB
2. Run Traffic_Portal locally (make sure the Traffic_Ops url point to CIAB traffic ops url)
3. Ensure protractor and webdriver-manage are installed.
4. Run webdriver-manager start
5. Run protractor config.js from traffic_control/traffic_portal/test/end-to-end/ folder
6. All 73 specs should pass. `73 specs, 0 failures`

## If this is a bug fix, what versions of Traffic Control are affected?


- master (d04111c)

## The following criteria are ALL met by this PR


- [x] This does not include a database migration


## Additional Information
